backend/routes/create_active_order.py

```diff
@@ -1,24 +1,165 @@
-"""Helper functions to create test orders"""
+"""Creates buy and sell orders"""
+
+from datetime import datetime, timezone
+from uuid import UUID, uuid4
+
+from fastapi import HTTPException
+from pydantic import BaseModel
+from starlette.status import HTTP_400_BAD_REQUEST
 
 from database import supabase_middleman
-from util import test_data
 
 
-def create_buy_order():
-    """
-    Inserst a test buy entry into active_buy_sell table.
+class ErrorResponse(BaseModel):
+    error_code: int
+    error_message: str
+    details: str = None  # Optional field
+
+
+def create_active_buy_sell_order(data: dict) -> str:
+    """Create a buy or sell order for a stock."""
+    buy_or_sell = data["buy_or_sell"]
+    price = data["price"]
+    expirey = data["expirey"]
+    quantity = data["quantity"]
+    stock_id = data["stockId"]
+    user_id = data["userId"]
+
+    expirey = expirey.replace(
+        "Z", "+00:00"
+    )  # Replace Z with +00:00 to make it ISO 8601 compliant
+    expirey = datetime.fromisoformat(expirey)  # Convert expirey to datetime object
+    if price < 1:
+        # raise_http_exception(
+        #     status_code=HTTP_400_BAD_REQUEST,
+        #     error_code=400,
+        #     error_message="Price must be greater than 0",
+        # )
+        return "Price must be greater than 0"
+    if quantity < 1:
+        # raise_http_exception(
+        #     status_code=HTTP_400_BAD_REQUEST,
+        #     error_code=400,
+        #     error_message="Quantity must be greater than 0",
+        # )
+        return "Quantity must be greater than 0"
+    if expirey < datetime.now(timezone.utc):
+        # raise_http_exception(
+        #     status_code=HTTP_400_BAD_REQUEST,
+        #     error_code=400,
+        #     error_message="Expirey date cannot be in the past",
+        # )
+        return "Expirey date cannot be in the past"
+    if not supabase_middleman.fetch_stock_price(stock_id):
+        # raise_http_exception(
+        #     status_code=HTTP_400_BAD_REQUEST,
+        #     error_code=400,
+        #     error_message="Stock not found",
+        # )
+        return "Stock not found"
```

Comment:
consider using an array instead of just returning a single error:

```suggestion
    errors = []
    if price < 1:
        errors.append("Price must be greater than 0")
    if quantity < 1:
        errors.append("Quantity must be greater than 0")
    if expirey < datetime.now(timezone.utc):
        errors.append("Expirey date cannot be in the past")
    if not supabase_middleman.fetch_stock_price(stock_id):
        errors.append("Stock not found")
    if errors:
        return ", ".join(errors)
```

Reply:
Avoid commenting out code and committing it to your source control system. When another developer sees your commented out code it can lead to confusion. Commented out code also tends to build up as developers are often scared to delete it thinking it must be important or related to another team member’s work.

Simply delete the line of code instead of commenting it out. Your repository has a history and getting the deleted line back should not be difficult if you need it again.

- https://bytedev.medium.com/code-comment-anti-patterns-and-why-the-comment-you-just-wrote-is-probably-not-needed-919a92cf6758